app/uk/gov/hmrc/teamsandrepositories/connectors/GithubConnector.scala

```diff
@@ -115,6 +120,61 @@ class GithubConnector @Inject()(
     )
   }
 
+  /**
+   * A combinator to augment a list of repositories with the branch protection policy of their default branches.
+   *
+   * This is performed after-the-fact and in-memory to avoid making an additional call-per-repository to the GitHub API
+   * (which would slow down the scheduled job and risk an altercation with GitHub's rate limiter.)
+   */
+  def withBranchProtectionPolicies(action: => Future[List[GhRepository]]): Future[List[GhRepository]] = {
+
+    def branchProtectionEnabled(repos: List[BranchProtectionQueryResponse.Repository]) =
+      repos
+        .map(r => (r.name, r.defaultBranchBranchProtection.isDefined))
```

Comment:
What do we mean by "branch protection enabled"?

Is there being any branch protection policy at all sufficient? (That's what I've punted on here until others weigh in)

Or would we like to define branch protection as (something like):

 - Requires review
 - Dismisses stale reviews
 - Potentially even ensure some checks have gone green

?

Reply:
We discussed and decided to capture:

- `requiresApprovingReviews`
- `dismissesStaleReviews`